presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -859,7 +861,24 @@ public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<
                         Sets.difference(ImmutableSet.copyOf(partitioning.getPartitionColumns()), allColumns)));
             }
             HivePartitioningHandle partitioningHandle = (HivePartitioningHandle) partitioning.getPartitioningHandle();
-            return new HiveBucketProperty(partitioning.getPartitionColumns(), partitioningHandle.getBucketCount(), ImmutableList.of());
+            List<String> partitionColumns = partitioning.getPartitionColumns();
+            if (partitionColumns.isEmpty() || !partitioningHandle.getTypes().isPresent()) {
+                return new HiveBucketProperty(
```

Comment:
Why do we need to use hive compatible bucket property if `partitionColumns.isEmpty() || !partitioningHandle.getTypes().isPresent()`?